java/flight/flight-core/src/main/java/org/apache/arrow/flight/auth2/BearerTokenAuthHandler.java

```diff
@@ -26,9 +26,12 @@
  * Partial implementation of ServerAuthHandler for bearer-token based authentication.
  */
 abstract class BearerTokenAuthHandler implements CallHeaderAuthenticator {
+
+  String bearerToken;
```

Comment:
There is a single CallHeaderAuthenticator that the server uses, so this logic does not work if you have multiple connections. Please add unit tests with multiple FlightClients connecting to the same server, both with differing credentials/identities and the same credentials.

Reply:
Done, added tests for both BasicCallHeaderAuthenticator and GeneratedBearerTokenAuthHandler.